src/audio/buffer.c

```diff
@@ -198,15 +200,17 @@ void comp_update_buffer_produce(struct comp_buffer *buffer, uint32_t bytes)
 	notifier_event(buffer, NOTIFIER_ID_BUFFER_PRODUCE,
 		       NOTIFIER_TARGET_CORE_LOCAL, &cb_data, sizeof(cb_data));
 
-	addr = buffer->stream.addr;
 
+#if !defined(CONFIG_LIBRARY)
```

Comment:
Would be better to just define buf_dbg as empty in testbench build.

Reply:
I'm trying but is there a way to avoid this error:

```
/home/singalsu/work/current/sof/sof/src/audio/buffer.c: In function ‘comp_update_buffer_produce’:
/home/singalsu/work/current/sof/sof/src/audio/buffer.c:182:8: error: variable ‘addr’ set but not used [-Werror=unused-but-set-variable]
  182 |  char *addr;
      |        ^~~~

```

Reply:
It can be fixed with this into buf_dbg() commands

```
-               ((char *)buffer->stream.r_ptr - addr) << 16 |
-               ((char *)buffer->stream.w_ptr - addr));
+               ((char *)buffer->stream.r_ptr - buffer->stream.addr) << 16 |
+               ((char *)buffer->stream.w_ptr - buffer->stream.addr));

```